### PR TITLE
Upgrade graphql-java to 20.1 and smallrye-graphql 2.1.3

### PR DESCRIPTION
**Pull request by jmartisk**

### CODE REVIEW

File: bom/application/pom.xml [hunk 1]
```diff
@@ -6127,6 +6128,12 @@
                 <groupId>org.eclipse.jgit</groupId>
                 <artifactId>org.eclipse.jgit</artifactId>
                 <version>${jgit.version}</version>
+	    </dependency>	
+
+            <dependency>
+                <groupId>com.graphql-java</groupId>
+                <artifactId>graphql-java</artifactId>
+                <version>${graphql-java.version}</version>
             </dependency>
```
Comment (phillip-kruger) [inline bom/application/pom.xml:6137]: Why is this needed ? Will this not be pulled in from the smallrye dep ?

**Review by phillip-kruger — COMMENTED** [review]
Seems like some of the CI failures is related. Can you check ? Thanks :)

File: bom/application/pom.xml (hunk 1 above)
Comment (jmartisk) [reply bom/application/pom.xml:6137]: This is to satisfy the enforcer plugin. Vert.x graphql extension still depends on graphql-java 19.2, so we need to get the convergence aligned. When something is pulled in by two different libraries, it's generally a good idea to manage the version manually.

**Comment by jmartisk** [comment]
> Seems like some of the CI failures is related. Can you check ? Thanks :)

Yeah I had forgotten a few changes, added them now. Hopefully it's correct now.

File: bom/application/pom.xml [hunk 2]
```diff
@@ -6127,6 +6128,12 @@
                 <groupId>org.eclipse.jgit</groupId>
                 <artifactId>org.eclipse.jgit</artifactId>
                 <version>${jgit.version}</version>
+	    </dependency>
```
Comment (gsmet) [inline bom/application/pom.xml:6131]: Indentation is off.

File: bom/application/pom.xml (hunk 1 above)
Comment (gsmet) [reply bom/application/pom.xml:6137]: Thanks for the explanation but please keep the graphql-java version with the SmallRye GraphQL one and add a comment at the end of the line of the first one saying both should be kept in sync. Otherwise we will forget about it.

**Review by gsmet — CHANGES_REQUESTED** [review]
Failures are unrelated but I spotted a few minor things. Could you take care of them?

File: bom/application/pom.xml (hunk 1 above)
Comment (jmartisk) [reply bom/application/pom.xml]: Added a comment. I put them together inside the `<dependencyManagement>` but I don't quite like putting the `<properties>` together because it would look weird to have one different property inside ~20 properties that start with `smallrye`

**Review by gsmet — APPROVED** [review]
